vend.js

```diff
@@ -171,8 +171,9 @@ var sendRequest = function(options, args, connectionInfo, callback, retryCounter
       return Promise.reject(e.statusCode + ' ' + e.response.body); // TODO: throw unknown errors but reject well known errors?
     })
     .catch(function(e) {
-      console.error('vend.js - sendRequest - An unexpected error occurred: ', e);
-      throw e; // TODO: throw unknown errors but reject well known errors?
+      //console.error('vend.js - sendRequest - An unexpected error occurred: ', e);
+      return Promise.reject(e.statusCode + ' ' + e.response.body);
```

Comment:
First of all, I can't say that I fully remember why I hesitated to use `reject` this here from the very beginning!

@armandchard - I would be ok with `return Promise.reject(e);` here instead because there is no guarantee (I think) that the kind of error that reaches this block has `.statusCode` or `.response.body` as it properties.

I'm already capturing well known errors such as `AuthNError` and `ClientError` in the previous catch blocks so if you ran into a well known error during your tests that you want to reject with said properties then please define and use it. Thoughts?